src/overlays/actors/ovl_Boss_Dodongo/z_boss_dodongo.c

```diff
@@ -51,311 +44,1718 @@ const ActorInit Boss_Dodongo_InitVars = {
+                                           { -1390.0f, 0.0f, -3304.0f },
+                                           { -890.0f, 0.0f, -2804.0f } };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Dodongo/func_808C32F4.s")
+                    Vec3f D_808CA538[] = { { -890.0f, 0.0f, -2804.0f },
+                                           { -390.0f, 0.0f, -3304.0f },
+                                           { -890.0f, 0.0f, -3804.0f },
+                                           { -1390.0f, 0.0f, -3304.0f } };
+                    if (this->unk_1A2 == 0) {
+                        phi_v0_3 = &D_808CA538[this->unk_1A0];
+                    } else {
+                        phi_v0_3 = &D_808CA508[this->unk_1A0];
+                    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Dodongo/func_808C3704.s")
+                    Math_SmoothStepToF(&this->cameraEye.x, phi_v0_3->x, 0.2f, this->unk_204 * 20.0f, 0.0f);
+                    Math_SmoothStepToF(&this->cameraEye.y, player->actor.world.pos.y + 30.0f, 0.1f,
+                                       this->unk_204 * 20.0f, 0.0f);
+                    Math_SmoothStepToF(&this->cameraEye.z, phi_v0_3->z, 0.1f, this->unk_204 * 20.0f, 0.0f);
+                    Math_SmoothStepToF(&this->unk_204, 1.0f, 1.0f, 0.02f, 0.0f);
+                } else {
+                    f32 temp;
+                    if (this->unk_1A2 == 0) {
+                        this->unk_230 += 0.01f;
+                    } else {
+                        this->unk_230 -= 0.01f;
+                    }
+                    Math_SmoothStepToF(&this->unk_22C, 220.0f, 0.1f, 5.0f, 0.1f);
+                    temp = sinf(this->unk_230);
+                    tempSin = temp * this->unk_22C;
+                    temp = cosf(this->unk_230);
+                    tempCos = temp * this->unk_22C;
+                    Math_SmoothStepToF(&this->cameraEye.x, this->actor.world.pos.x + tempSin, 0.2f, 50.0f, 0.0f);
+                    Math_SmoothStepToF(&this->cameraEye.y, this->actor.world.pos.y + 20.0f, 0.2f, 50.0f, 0.0f);
+                    Math_SmoothStepToF(&this->cameraEye.z, this->actor.world.pos.z + tempCos, 0.2f, 50.0f, 0.0f);
+                    Math_SmoothStepToF(&this->unk_23C, 0.0f, 0.2f, 0.01f, 0.0f);
+                    /*Math_SmoothStepToF(&this->unk_22C, 220.0f, 0.1f, 5.0f, 0.1f);
+                    tempSin = this->unk_22C * sinf(this->unk_230);
+                    tempCos = this->unk_22C * cosf(this->unk_230);
+                    tempSin += this->actor.world.pos.x;
+                    Math_SmoothStepToF(&this->cameraEye.x, this->actor.world.pos.x + tempSin, 0.2f, 50.0f, 0.0f);
+                    Math_SmoothStepToF(&this->cameraEye.y, this->actor.world.pos.y + 20.0f, 0.2f, 50.0f, 0.0f);
+                    Math_SmoothStepToF(&this->cameraEye.z, this->actor.world.pos.z + tempCos, 0.2f, 50.0f, 0.0f);
+                    Math_SmoothStepToF(&this->unk_23C, 0.0f, 0.2f, 0.01f, 0.0f);*/
```

Comment:
probably remove unless you think itll be helpful for whoever tries to match this. my guy feeling says it can prob go